FoodBookApp/UI/Views/Login/LoginViewModel.swift

```diff
@@ -6,13 +6,38 @@
 //
 
 import Foundation
+import FirebaseAuth
+import GoogleSignIn
 
 @MainActor
-@Observable final class LoginViewModel {
+@Observable 
+final class LoginViewModel {
     
-    func signInGoogle() async throws {
+    var showAlert = false
+    var errorMsg: String = ""
+    
+    func signInGoogle() async {
         let helper = SignInGoogleHelper()
-        let tokens = try await helper.signIn()
-        let authDataResult = try await AuthService.shared.signInWithGoogle(tokens: tokens)
+        
+        do {
+            let tokens = try await helper.signIn()
+            let authDataResult = try await AuthService.shared.signInWithGoogle(tokens: tokens)
+        } catch GIDSignInError.canceled {
+            errorMsg = "You canceled the sign-in flow. If you wish to sign in, press the button again and complete the authentication."
```

Comment:
What does "cancel the sign-in flow" mean?

Reply:
It's when they press the 'Cancel' button on the browser that pops up for the login!